### PR TITLE
Add new routes to auto and refactor 

### PR DESCRIPTION
**Pull request by josieammer**
* Fixes bug in ArmToLocationCommand causing invalid positions to be commanded
* connects up the selectable chooser for scoring position to the arm commands
* adds Node2, Node8 auto routines
* changes paths for left, right, Node2, Node8 routines out of the community and (optionally) onto ramp
* rename some AutonomousChooser methods to make intent more clear
* reorganize AutonomousChooser methods to canonical ordering (public then private, alphabetical order)

### CODE REVIEW

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 1]
```diff
@@ -158,57 +171,88 @@ else if (this.subsystems.getEveryBotPickerSubsystem() != null) {// cube uses upt
...
+    private Command getArmPositionRoutine (ScoringPosition height){
         SequentialCommandGroup command = new SequentialCommandGroup();
-        command.addCommands(getScoreRoutine(NodePosition));
-        command.addCommands(new DriveTrajectoryCommand(subsystems.getDriveTrainSubsystem(), Trajectory));
+        if (height == ScoringPosition.SCORE_HIGH){
+            command.addCommands(new ArmToLocationCommand(subsystems.getArmSubsystem(), ArmToLocationCommand.ArmLocation.ARM_HIGH_SCORE));
+        }
+        else if (height == ScoringPosition.SCORE_MIDDLE){
+            command.addCommands(new ArmToLocationCommand(subsystems.getArmSubsystem(), ArmToLocationCommand.ArmLocation.ARM_MED_SCORE));
+        }
```
Comment (mmcadamsms) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:186]: So we won't ever try for low?  Seems about right.

Also don't actually need to wrap the ArmToLocationCommand in a SequentialCommandGroup but its only a bit of extra memory.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 2]
```diff
@@ -158,57 +171,88 @@ else if (this.subsystems.getEveryBotPickerSubsystem() != null) {// cube uses upt
...
      * @param DoBalance the enum for wether the balance routine should be run
-     * @return
+     * @return command
      */
-    public Command getBalanceRoutine (AutonomousBalance DoBalance){
+    private Command getBalanceRoutine (AutonomousBalance DoBalance, Trajectory toRampTrajectory){
         SequentialCommandGroup command = new SequentialCommandGroup();
         if (DoBalance == AutonomousBalance.DO_BALANCE){
-            command.addCommands(new DriveTrajectoryCommand(subsystems.getDriveTrainSubsystem(), trajectories.BehindToOntoRampTrajectory));
+            command.addCommands(new DriveTrajectoryCommand(subsystems.getDriveTrainSubsystem(), toRampTrajectory));
```
Comment (mmcadamsms) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:198]: So wanted to make sure this is intentional, in the event that DoBalance is not AutonomousBalance.DO_BALANCE then the 'toRampTrajectory' will not be completed.  Is this the intent to not do that bit of trajectory?

**Review by mmcadamsms — APPROVED** [review]
Couple of questions ... seems fine, just need to test it all.

File: smokey/src/main/java/frc/robot/control/Trajectories.java [hunk 3]
```diff
@@ -10,14 +10,18 @@
 
 public class Trajectories {
     public Pose2d Node1Position;
+    public Pose2d Node2Position;
```
[the remark this answers is not in the document]
Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/control/Trajectories.java:13]: so just looked it up, 'friend' is a C++ thing ... would need to go with something else

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java (hunk 1 above)
Comment (josieammer) [reply smokey/src/main/java/frc/robot/control/AutonomousChooser.java:186]: Correct.  We don't need low height in auto.

Comment (josieammer) [reply smokey/src/main/java/frc/robot/control/AutonomousChooser.java:186]: removed SequentialCommandGroup

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java (hunk 2 above)
Comment (josieammer) [reply smokey/src/main/java/frc/robot/control/AutonomousChooser.java:198]: Correct. When we are not balancing, there is a different team balancing and we want to stay out of their way.

File: smokey/src/main/java/frc/robot/control/Trajectories.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (josieammer) [reply smokey/src/main/java/frc/robot/control/Trajectories.java:13]: changed all members to private and created getters.

**Pull request merged**